pandas/core/generic.py

```diff
@@ -5620,7 +5620,7 @@ def rank(self, axis=0, method='average', numeric_only=None,
             raise NotImplementedError(msg)
 
         def ranker(data):
-            ranks = algos.rank(data.values, axis=axis, method=method,
+            ranks = algos.rank(data.values.copy(), axis=axis, method=method,
```

Comment:
do this in ``pandas/core/algorithms.py`` instead (even better would be to change the actual cython code to avoid the need for a copy here)